system/modules/isotope/library/Isotope/Model/Address.php

```diff
@@ -203,7 +203,21 @@ public function getTokens($arrFields = null)
 
                 list($country, $subdivion) = explode('-', $this->subdivision);
 
-                $arrTokens['subdivision']      = $arrSubdivisions[strtolower($country)][$this->subdivision];
+                foreach ($arrSubdivisions[strtolower($country)] as $v) {
+                    // Support multi-dimensional arrays (e.g. subdivisions of United Kingdom)
+                    if (is_array($v)) {
+                        foreach ($v as $vv) {
+                            if ($vv[$this->subdivision]) {
+                                $arrTokens['subdivision'] = $vv[$this->subdivision];
+                                break;
+                            }
+                        }
+                    } else {
+                        $arrTokens['subdivision'] = $arrSubdivisions[strtolower($country)][$this->subdivision];
```

Comment:
This can't be right. We're `foreach`ing the array but then not use any of it's foreach-value in the loop? 

Did you test this? On the first spot it seems wrong to have two `foreach`?